tests/apitest/src/CoreGraphics/CGDisplay.cs

```diff
@@ -0,0 +1,99 @@
+﻿using NUnit.Framework;
+using System;
+
+#if !XAMCORE_2_0
+using MonoMac.AppKit;
+using MonoMac.ObjCRuntime;
+using MonoMac.CoreImage;
+using MonoMac.CoreGraphics;
+using CGSize = System.Drawing.SizeF;
+using CGRect = System.Drawing.RectangleF;
+#else
+using AppKit;
+using ObjCRuntime;
+using CoreImage;
+using CoreGraphics;
+#endif
+
+namespace Xamarin.Mac.Tests
+{
+	[TestFixture]
+	public class CGDisplayTests
+	{
+		int mainDisplayId;
+
+		[SetUp]
+		public void Setup ()
+		{
+			mainDisplayId = CGDisplay.MainDisplayID;
+		}
+
+		[Test]
+		public void CGDisplayShouldGetDimensions ()
+		{
+			var bounds = CGDisplay.GetBounds (mainDisplayId);
+			Assert.IsFalse (bounds.IsNull ());
+			Assert.IsFalse (bounds.IsEmpty);
+
+			var width = CGDisplay.GetWidth (mainDisplayId);
+			Assert.Greater (0, width);
+
+			var height = CGDisplay.GetHeight (mainDisplayId);
+			Assert.Greater (0, height);
+		}
+
+		[Test]
+		public void CGDisplayCursorMethods ()
+		{
+			int error;
+			error = CGDisplay.HideCursor (mainDisplayId);
+			Assert.Equals (0, error);
+
+			error = CGDisplay.ShowCursor (mainDisplayId);
+			Assert.Equals (0, error);
+
+			error = CGDisplay.MoveCursor (mainDisplayId, new CGPoint (0, 0));
+			Assert.Equals (0, error);
+		}
+
+		[Test]
+		public void CGDisplayShouldSetAndRestoreDisplayTransfer ()
+		{
+			Exception exception = null;
+
+			try {
+				var error = CGDisplay.SetDisplayTransfer (mainDisplayId, 0, .5f, .5f, 0, .5f, .5f, 0, .5f, .5f);
```

Comment:
We talked about this on Slack, I think this makes more sense as a manual test.

Anyway, mouse/display stuff feels like it might act funny on vm / bots. Last thing I want is a flakey test (even if it's Apple's stuff that is acting weird).